### ISSUE TITLE
Unreleased Rack Version Compatibility Issue

### ISSUE THREAD

**Issue by tomprats** [rails/rails#23007]
I'm using oath2 in an application (through Twitter, Facebook, and Instagram) and I was hoping to upgrade to Rails 5 (beta). When doing so, I ran into a dependency problem. It looks like version 2 of Rack isn't released yet, but is trusted enough to be used in Rails 5.

Are the only solutions to this problem:
- Not updating to Rails 5
- Forking Oath2 and updating its dependencies
  - I just created an issue here https://github.com/intridea/oauth2/issues/229
- Forking Rails 5 and downgrading its dependencies 
  - Definitely seems like the worst choice, so feel free to ignore

```
Bundler could not find compatible versions for gem "rack":
  In Gemfile:
    rails (>= 0) ruby depends on
      railties (= 5.0.0.beta1) ruby depends on
        actionpack (= 5.0.0.beta1) ruby depends on
          rack (~> 2.x) ruby

    rails (>= 0) ruby depends on
      railties (= 5.0.0.beta1) ruby depends on
        actionpack (= 5.0.0.beta1) ruby depends on
          rack-test (~> 0.6.3) ruby depends on
            rack (>= 1.0) ruby

    omniauth-facebook (~> 3.0) ruby depends on
      omniauth-oauth2 (~> 1.2) ruby depends on
        oauth2 (~> 1.0) ruby depends on
          rack (~> 1.2) ruby

    rack (>= 0) ruby
```

**Comment by nateberkopec** [comment]
> Forking Oath2 and updating its dependencies

Yup, you should do this and use your fork until intridea/oauth2#229 is merged.

**Comment by maclover7** [comment]
Few things:
-Agree with Nate here.
-Rails 5 is currently beta software, so it's fine for it to be depending on another beta of Rack. Before 5.0 is shipped, Rack 2.0 will hopefully be shipped as well.
-Finally, we reserve the issue tracker for bugs only.  Please continue this discussion on oauth2's repo.
